[13.0] [MIG] fieldservice_skill

brian10048 opened the pull request:
Migrate fieldservice_skill to 13.0

For #354

Commit:
[ADD] fieldservice_skill

max3903 committed:
[ADD] Security

max3903 committed:
[IMP] View

osi-scampbell committed:
[IMP] Various

osi-scampbell committed:
[IMP] Connect fsm.skill to fsm.template (#113)

* [IMP] Connect fsm.skill to fsm.template

osi-scampbell committed:
[IMP] Smart button

osi-scampbell committed:
[IMP] Added smartbutton

osi-scampbell committed:
[IMP] Update to 11.0

osi-scampbell committed:
[IMP] Remove line

osi-scampbell committed:
[IMP] Instructions to Todo

max3903 committed:
[FIX] fieldservice_skill

max3903 committed:
[FIX] fieldservice_skill - Call super

osi-scampbell committed:
[IMP] FSM Skill Bug Fix

max3903 committed:
[FIX] fieldservice_skill: duplicate record

murtuzasaleh committed:
[MIG][WIP][12.0] fieldservice_skill

brian10048 committed:
[IMP] UI

minor UI improvements

max3903 committed:
[IMP] fieldservice_skill

marylla committed:
Added translation using Weblate (German)

marylla committed:
Translated using Weblate (German)

Currently translated at 48.0% (12 of 25 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice_skill
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice_skill/de/

marylla committed:
Translated using Weblate (German)

Currently translated at 100.0% (25 of 25 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice_skill
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice_skill/de/

Commit:
Added translation using Weblate (Spanish (Chile))

Commit:
Translated using Weblate (Spanish (Chile))

Currently translated at 88.0% (22 of 25 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice_skill
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice_skill/es_CL/

marcelsavegnago committed:
Added translation using Weblate (Portuguese (Brazil))

max3903 committed:
[ADD] Spanish (es) translation

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (25 of 25 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice_skill
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice_skill/pt_BR/

fcolus committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (25 of 25 strings)

Translation: field-service-12.0/field-service-12.0-fieldservice_skill
Translate-URL: https://translation.odoo-community.org/projects/field-service-12-0/field-service-12-0-fieldservice_skill/pt_BR/

brian10048 committed:
[IMP] fieldservice_skill: black, isort, prettier

brian10048 committed:
[MIG] fieldservice_skill: Migration to 13.0

bodedra commented:
This PR depends on "hr_skill" module and its functionality come in Odoo13 hr_skills module.

![Firefox_Screenshot_2020-07-21T19-02-24 423Z](https://user-images.githubusercontent.com/14229503/88095536-23640f80-cb4a-11ea-9660-6216f526f521.png)

So change module dependency from "hr_skill" to "hr_skills"

bodedra commented on fieldservice_skill/__manifest__.py:
```diff
@@ -0,0 +1,25 @@
+    "license": "AGPL-3",
+    "author": "Open Source Integrators, Odoo Community Association (OCA)",
+    "website": "https://github.com/OCA/field-service",
+    "depends": ["hr_skill", "fieldservice"],
```
```suggestion
    "depends": ["hr_skills", "fieldservice"],
```

brian10048 commented:
> 
> 
> This PR depends on "hr_skill" module and its functionality come in Odoo13 hr_skills module.
> 
> ![Firefox_Screenshot_2020-07-21T19-02-24 423Z](https://user-images.githubusercontent.com/14229503/88095536-23640f80-cb4a-11ea-9660-6216f526f521.png)
> 
> So change module dependency from "hr_skill" to "hr_skills"

I updated the dependency and updated an inherited view name.

Will a migration script be needed because of this module change?

bodedra commented:
@brian10048 I don't think we need a migration script because it has same model name "hr.skill" So we are safe.

cc @max3903 @dreispt

RLeeGML commented:
@max3903 @osi-scampbell 

Closed my PR, I missed that this was already open. Same questions as before.

The dependency hr_skill from OCA/hr has been changed to hr_skills from odoo/odoo. The old module had a color field on hr.skill model, while the new one does not. Does this need to be added back in?

The Usage file indicates that the worker list on fsm.order will be filtered by worker skill (as well as location) but there's no functionality to filter by worker skill in the v12 module. Should this be added in?

murtuzasaleh commented on fieldservice_skill/views/hr_skill.xml:
```diff
@@ -0,0 +1,10 @@
+    <menuitem
+        id="menu_hr_skill"
+        name="Skills"
+        action="hr_skills.hr_skill_view_form"
```
@brian10048 Can you please improve the action name or create new action of skills and bind it here?

dreispt commented:
🚨 Red build!

brian10048 commented:
@RLeeOSI I saw your PR at [https://github.com/brian10048/field-service/pull/16](https://github.com/brian10048/field-service/pull/16), but I wonder if we should modify the v12 behavior to instead open the Skill Type view as it provides a more complete overview of default skills configuration like I recently updated here.

After looking at this more, I wonder if we should update the `fsm.person.skill` to more or less mimic [`hr.employee.skill`](https://github.com/odoo/odoo/blob/e9c54f6b3d5c36b658308e1900ab68dcc9de0357/addons/hr_skills/models/hr_skills.py#L16:30)
The Odoo skill module has a level progress feature that can replace the level priority style field on fsm.person.skill

I don't mind making the changes, just want to make sure we are all on the same page first.  @dreispt @murtuzasaleh  Let me know what you guys think as well

RLeeGML commented:
@brian10048 I agree, it looks like the Odoo module intends to use hr.skill.type as the main model for skills (hence why there's a menuitem for it and not hr.skill). @max3903 and @osi-scampbell your input here would be greatly appreciated, and on the earlier comment [here](https://github.com/OCA/field-service/pull/602#issuecomment-706433962)

max3903 commented:
> The dependency hr_skill from OCA/hr has been changed to hr_skills from odoo/odoo. The old module had a color field on hr.skill model, while the new one does not. Does this need to be added back in?

Yes

> The Usage file indicates that the worker list on fsm.order will be filtered by worker skill (as well as location) but there's no functionality to filter by worker skill in the v12 module. Should this be added in?

Yes

brian10048 committed:
[IMP] fieldservice_skill: Add back the color field

brian10048 committed:
[IMP] fieldservice_skill: Filter order workers

dreispt commented:
@brian10048 Lint checks are failing.
Can you check if you have [pre-commit configured](https://github.com/OCA/maintainer-quality-tools/#qa-tools-for-odoo-maintainers-mqt) on your local v13 repository?

brian10048 commented:
@RLeeOSI - to continue the conversation here for the others to see...
> @brian10048 do we need the migration script? The module name changes but the model is still "hr.skill"

Good catch.  The Odoo module and file name says hr_skills, but the model name is hr.skill like it was for the OCA - So this particular migration script is not relevant.  I think one would be required for those migrating from the OCA skills module to Odoo skills module, but that is probably not in the scope of field-service ??

Also, the domain filter I have added on the orders view to filter workers by skill is not working correctly.  I need to fix this as well and will fix those lint errors

bodedra commented:
@brian10048 I agree with you.

There is no need migration script or any special treatment. fieldservice_skill will depends on Odoo's hr_skills module.

Would you please fix linting erros.

bodedra commented on fieldservice_skill/views/fsm_order.xml:
```diff
@@ -0,0 +1,24 @@
+                    options="{'color_field': 'color'}"
+                />
+            </field>
+            <field name="skill_worker_ids" invisible="1" />
```
```suggestion
```

bodedra commented on fieldservice_skill/views/fsm_order.xml:
```diff
@@ -0,0 +1,24 @@
+                    name="skill_ids"
+                    widget="many2many_tags"
+                    options="{'color_field': 'color'}"
+                />
```
```suggestion
                />
                <field name="skill_worker_ids" invisible="1" />
```

bodedra reviewed (commented):
minor one

brian10048 committed:
[IMP] fieldservice_skill: Add Test

brian10048 commented:
![image](https://user-images.githubusercontent.com/25710115/97249368-13f40280-17da-11eb-928d-59850eafdc68.png)

bodedra reviewed (approved):
Functional reviewed. :+1: 

Non-blocking comment.

I have found one minor gap.

When we set/update `Required Skills`, `Assigned To` list does not filter until save record.

Filter logic has implemented on compute field. So if we add onchange on `Required Skills` with same logic, `Assigned To` list will be filter without save record.

bodedra reviewed (approved):
Awesome. LG :+1:

brian10048 commented:
![Screenshot from 2020-10-28 16-37-08](https://user-images.githubusercontent.com/25710115/97493812-e58e3880-193b-11eb-991b-55ad1c6d9491.png)

I am implementing a cleaner look for the worker skills to match the Odoo v13 skills look for employees.

I will have that ready for a final review soon

brian10048 committed:
[IMP] fieldservice_skill: Skill views

Adapted FSM Person Skills to mimic Odoo v13 behavior of Employee Skills.
- Skill progress level instead of priority
- Updated views on FSM Person form

bodedra commented:
@brian10048 Would you please fix this?

![Screenshot from 2020-10-29 16-17-56](https://user-images.githubusercontent.com/14229503/97642490-8515fe80-1a02-11eb-9e64-a63271190f97.png)

brian10048 commented:
> 
> 
> @brian10048 Would you please fix this?
>

Ready for final review @bodedra

bodedra reviewed (approved):
@brian10048 Functional reviewed. works great! :+1: 

+1 for beautiful view.

![Screenshot from 2020-10-30 11-06-44](https://user-images.githubusercontent.com/14229503/97741801-8bf34e80-1aa0-11eb-843f-518fa4e6b50d.png)

bodedra commented:
@max3903 @dreispt would you please give your blessing to this PR!

brian10048 commented:
Thanks @bodedra, we can thank Odoo for the design of the view. I just thought it should remain cohesive here!

marcelsavegnago reviewed (approved):
LGTM. It's really pretty. Congratulations.

max3903 commented:
/ocabot merge nobump
